src/main/java/com/burgstaller/okhttp/DefaultProxyCacheKeyProvider.java

```diff
@@ -0,0 +1,23 @@
+package com.burgstaller.okhttp;
+
+import java.net.Proxy;
+
+/**
+ * The default version of the cache key provider, which simply calls the java.net.Proxy.toString() method to generate key.
+ */
+public final class DefaultProxyCacheKeyProvider implements CacheKeyProvider<Proxy> {
+    @Override
+    public boolean applyToProxy() {
+        return true;
+    }
+
+    /**
+     *
+     * @param proxy
+     * @return the cache key.
+     */
+    @Override
+    public String getCachingKey(Proxy proxy) {
+        return proxy!=null?proxy.toString():null;
```

Comment:
it would be super nice, if you would keep to the existing formatting
```suggestion
        return proxy != null ? proxy.toString() : null;
```

Reply:
done